### ISSUE TITLE
publishToNexus name and also credential properties

### ISSUE THREAD

**Issue by Vampire** [marcphilipp/nexus-publish-plugin#14]
As "nexus" is just the repository management software used at sonatype, I set the `repositoryName` to `sonatype`.
Now I wonder, shouldn't maybe the `publishToNexus` task maybe called `"publishTo${repositoryName.capitalize()}"`?
And also the credential propeties `${repositoryName}Username` and `${repositoryName}Password`?

**Comment by Vampire** [comment]
And of course also `initializeNexusStagingRepository` => `initialize${repositoryName.capitalize()}StagingRepository`

**Comment by marcphilipp** [comment]
Good questions! Since this plugin only supports a single repository, I think making the task names and config properties "dynamic" would add more confusion than make it clearer. WDYT?

**Comment by Vampire** [comment]
What do you mean by "single repository"?
You mean within one build?
Maybe this should be changed, so you can use it to release internally and to sonatype within one run. :-)

But even if not, actually that is what I would have expected and would be consistent to Gradle standard behavior.
You create a publication called `Xxx` and a repository called `Yyy`, then you get a task `publishXxxPublicationToYyyRepository`.
And if the properties also adjust, you can e. g. have the properties simply set in a properties file if the build supports deploying to multiple repos depending on some project property (as long as the plugin does not support it itself), without the property names adjusting, you have to have additional logic in the build script that sets the values on the extension, depending on the same project property.

I think most users will go with the default anyway so will not be confused.
And the ones preferring other names would maybe be happy with having all the names adjust.
And if you (hopefully) decide in the future to also support multiple targets, it is probably necessary then latest anyway.

Actually I wondered already that you do not support multiple repos as you store the staging repo id in a map, associated to the server URL.

**Comment by marcphilipp** [comment]
What we'd really need is a way to define a custom repository type or a way to extend `MavenArtifactRepository`.

Since that isn't currently possible, IIUC you'd prefer a config like the following, right?

```gradle
nexusRepositories {
    sonatype {
        serverUrl = uri("https://oss.sonatype.org/service/local/") // no default
        snapshotRepositoryUrl = uri("https://oss.sonatype.org/content/repositories/snapshots/") // no default
        username = "your-username" // defaults to project.properties["sonatypeUsername"]
        password = "your-password" // defaults to project.properties["sonatypePassword"]
    }
    myNexus {
        serverUrl = uri("https://your-server.com/staging")
        snapshotRepositoryUrl = uri("https://your-server.com/snapshots") // no default
        username = "your-username" // defaults to project.properties["myNexusUsername"]
        password = "your-password" // defaults to project.properties["myNexusPassword"]
    }
}

**Comment by marcphilipp** [comment]
@Vampire ?

**Comment by vlsi** [comment]
@marcphilipp , i think the above config would work.

My case is:
1) https://oss.sonatype.org/ for Maven Central
2) https://repository.apache.org/ for ASF repository

Apparently I can't use the same credentials there so `nexusPassword` property is not good enough.

On the other hand, I don't really need to push to multiple Nexus repositories at the same time yet (so I don't probably need `publishXxxPublicationToYyyRepository` yet)

**Comment by Vampire** [comment]
Sorry for the late answer @marcphilipp, yes, something like that was what I had in mind.

**Comment by vlsi** [comment]
It looks like `publishToNexus` duplicates built-it `"publishAllPublicationsTo" + capitalize(repository.getName()) + "Repository"` task which is already present in Gradle.

Should `publishToNexus` be just removed from N-P-P?
See also https://github.com/gradle/gradle/issues/10132

**Comment by marcphilipp** [comment]
Looks like this was added in Gradle 5.3. I wasn't aware of that, thanks for the pointer! 👍

**Comment by marcphilipp** [comment]
@vlsi Regarding the `"publishAllPublicationsTo" + capitalize(repository.getName()) + "Repository"` task I'm undecided. I think I'll leave the shorter `publishTo$RepoName` in place for now.

**Comment by vlsi** [comment]
>Regarding the "publishAllPublicationsTo" + capitalize(repository.getName()) + "Repository" task I'm undecided. I think I'll leave the shorter publishTo$RepoName in place for now.

I though as follows:
1) Gradle has the feature embedded, so you might want to provide a way to avoid creation of the task that duplicates the function.
2) I thought you were on Gradle team, so you might have better chances than me to influence Gradle core so they rename longer task to a shorter name (see item 4 in https://github.com/gradle/gradle/issues/10132 )

**Comment by marcphilipp** [comment]
This plugin is compatible with Gradle 4.10.x and the new task is only available since 5.3. Thus I think there's some value in keeping it. While I'm part of the Gradle team I don't work on the build tool at the moment. I'll bring it up with @melix who has added the task but chances are we'll have to keep it for backwards compatibility. Thus, I see two options for this plugin: Remove the task or create it only for Gradle < 5.3 with the longer name.
